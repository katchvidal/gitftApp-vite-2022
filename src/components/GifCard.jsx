import React from 'react'

export const GifCard = ({ url, title, id }) => {
  return (
    <div className='card'> 
        <p> { title } </p>
        <img src={ url } />
    </div>
  )
}
