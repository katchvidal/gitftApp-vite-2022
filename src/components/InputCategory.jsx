import React, { useState } from 'react'

export const InputCategory = ({ onAddCategory }) => {
    const [inputValue, setInputValue] = useState('')

    const handleChange = ({ target }) => {
        setInputValue(target.value)
    }

    const handleSubmit = ( e ) => {
        e.preventDefault()
        if ( inputValue.trim().length <= 1 ) return;
        onAddCategory( inputValue )
        setInputValue('')
    }

    return (
        <>
            <form onSubmit={ handleSubmit }>
                <input
                    type='text'
                    value={inputValue}
                    onChange={handleChange}
                    placeholder='Introduce Categoria / Personaje / Serie a Buscar'
                />
                <button type='submit'> Buscar </button>
            </form>
        </>
    )
}
