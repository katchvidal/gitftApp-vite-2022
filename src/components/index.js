export * from './InputCategory';
export * from './GiftGrid';
export * from './GifCard';