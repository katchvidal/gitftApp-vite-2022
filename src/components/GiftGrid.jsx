import React, { useEffect, useState } from 'react'
import { getGifs } from './../lib/getGifs'
import { GifCard } from './GifCard'

export const GiftGrid = ({ category }) => {
    const [images, setImages] = useState([])
    const getImages = async () => {
        const newImages = await getGifs(10, category);
        setImages(newImages)
    }
    useEffect(() => {
        getImages();

    }, [])

    return (

        <>
            <h2> {category} </h2>
            <div className='card-grid'>

                {
                    images.map(({ id, title, url }) => (
                        <GifCard key={id} title={title} url={url} />
                    ))
                }
            </div>
        </>
    )
}
