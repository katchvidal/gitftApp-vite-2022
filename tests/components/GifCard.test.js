import { render } from "@testing-library/react"
import { GifCard } from "../../src/components/GifCard"
import { screen } from '@testing-library/react'

describe('Pruebas en Gif Card', () => { 
    const title = 'Pug'
    const url = 'https://google.com/pug.jpg'
    test('Debe hacer Match vs SnapShot', () => { 
        const { container } = render( <GifCard title={ title } url={ url } />)
        expect( container ).toMatchSnapshot();
     });
     test('Debe mostrar imagen con la url y texto alternativo', () => { 
        render( <GifCard title={ title } url={ url } />)
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
      });

      test('should title as string', () => { 
        render(<GifCard title={ title } url={ url } />);
        // screen.debug();
        expect ( screen.getByText ( title )).toBeTruthy();
       })
})

// Con la u actualizamos el snapshot