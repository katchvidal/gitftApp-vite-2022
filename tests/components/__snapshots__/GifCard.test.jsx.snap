// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Gif Card Debe hacer Match vs SnapShot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
       
      Pug
       
    </p>
    <img
      alt="Pug"
      src="https://google.com/pug.jpg"
    />
  </div>
</div>
`;
